src/anderson_darling.jl

```diff
@@ -9,18 +9,13 @@ abstract type ADTest <: HypothesisTest end
 
 function adstats(x::AbstractVector{T}, d::UnivariateDistribution) where T<:Real
     n = length(x)
-    y = sort(x)
-    μ = mean(y)
-    σ = std(y)
-    A² = convert(typeof(μ), -n)
+    A² = convert(Float64, -n)
     for i = 1:n
-        zi = (y[i] - μ)/σ
-        zni1 = (y[n - i + 1] - μ)/σ
-        lcdfz = logcdf(d, zi)
-        lccdfz = logccdf(d, zni1)
-        A² -= (2*i - 1)/n * (lcdfz + lccdfz)
+        lcdfz = logcdf(d, x[i])::Float64
```

Comment:
No need for type assertions. The formula used to compute `A²` should give a `Float64` anyway due to `(i+i-1)/n`. Also assertions will fail if the type doesn't match, they don't perform the conversion automatically, so that's a lose-lose solution.